fileblock/src/inode.rs

```diff
@@ -0,0 +1,17 @@
+
+/// an description of the type of file object pointed to by an Inode
+enum FileType {
+    /// a regular file
+    RegularFile,
+    /// a directory tree containing 1 or more regular files or directories
+    Directory
+}
+
+/// index node containing most of the interesting bits about a file object on disk
+struct Inode {
+    inumber: u32,
+    ftype: FileType,
+    fsize: u64,
```

Comment:
Comment here on units (i.e. is this bytes or blocks or something else)

Reply:
We might _need_ an exact bytes count to know where to stop reading at the last block.

That is... unless you can imagine another way of reading the data blocks for the read request and not returning free file padded to the nearest block.